### PR TITLE
[ja] update kubectl index

### PR DESCRIPTION
**Pull request by s-kawamura-w664**
This PR updates content/ja/docs/reference/kubectl/_index.md.

Update Summary:
- "In-cluster authentication and namespace overrides" section is added.
- The position is changed in operations table.
- The resource types table is updated.

Fixes: #40136

### CODE REVIEW

**Comment by atoato88** [comment]
The target page on preview site:
- https://deploy-preview-40138--kubernetes-io-main-staging.netlify.app/ja/docs/reference/kubectl/

File: content/ja/docs/reference/kubectl/_index.md [hunk 1]
```diff
@@ -66,6 +66,32 @@ kubectl [command] [TYPE] [NAME] [flags]
...
 
+## クラスター内認証と名前空間のオーバーライド {#in-cluster-authentication-and-namespace-overrides}
+
+デフォルトでは、`kubectl`は最初にPod内で動作しているか、つまりクラスター内で動作しているかを判断します。まず、`KUBERNETES_SERVICE_HOST`と`KUBERNETES_SERVICE_PORT`の環境変数を確認し、サービスアカウントのトークンファイルが`/var/run/secrets/kubernetes.io/serviceaccount/token`に存在するかどうかを確認します。3つともクラスター内で見つかった場合、クラスター内認証とみなされます。
+
+後方互換性を保つため、クラスター内認証時に`POD_NAMESPACE`環境変数が設定されている場合には、サービスアカウントトークンのデフォルトの名前空間が上書きされます。名前空間のデフォルトに依存しているすべてのマニフェストやツールは、この影響を受けます。
+
+**`POD_NAMESPACE`環境変数**
+
+`POD_NAMESPACE`環境変数が設定されている場合、名前空間に属するリソースのCLI操作は、デフォルトで環境変数の値になります。これは、Podが名前空間に属するリソースであり、コマンドで名前空間が指定されていないためです。`kubectl api-resources`の出力を見て、リソースが名前空間に属するかどうかを判断してください。
```
Comment (atoato88) [inline content/ja/docs/reference/kubectl/_index.md:77]: There is the sentence of followings on [`en` content](https://deploy-preview-40138--kubernetes-io-main-staging.netlify.app/docs/reference/kubectl/):
```
For example, if the variable is set to seattle, kubectl get pods would return pods in the seattle namespace. 
```

It seems the sentence is not translated into `ja`.

**Review by atoato88 — COMMENTED** [review]
I left 1 comment.
Please check it.

File: content/ja/docs/reference/kubectl/_index.md [hunk 2]
```diff
@@ -66,6 +66,32 @@ kubectl [command] [TYPE] [NAME] [flags]
 
 ヘルプが必要な場合は、ターミナルウィンドウから`kubectl help`を実行してください。
 
+## クラスター内認証と名前空間のオーバーライド {#in-cluster-authentication-and-namespace-overrides}
+
+デフォルトでは、`kubectl`は最初にPod内で動作しているか、つまりクラスター内で動作しているかを判断します。まず、`KUBERNETES_SERVICE_HOST`と`KUBERNETES_SERVICE_PORT`の環境変数を確認し、サービスアカウントのトークンファイルが`/var/run/secrets/kubernetes.io/serviceaccount/token`に存在するかどうかを確認します。3つともクラスター内で見つかった場合、クラスター内認証とみなされます。
```
Comment (nasa9084) [inline content/ja/docs/reference/kubectl/_index.md:71]:
```suggestion
デフォルトでは、`kubectl`は最初にPod内で動作しているか、つまりクラスター内で動作しているかどうかを判断します。まず、`KUBERNETES_SERVICE_HOST`と`KUBERNETES_SERVICE_PORT`の環境変数を確認し、サービスアカウントのトークンファイルが`/var/run/secrets/kubernetes.io/serviceaccount/token`に存在するかどうかを確認します。3つともクラスター内で見つかった場合、クラスター内認証とみなされます。
```
個人的な感覚かもしれないですが、「Aか、Bか」だけだと"A or B"っぽく見えて、「つまり」が違和感で読み返す必要があったので、後者を「〜かどうか」にした方が読みやすい、と思います

File: content/ja/docs/reference/kubectl/_index.md [hunk 3]
```diff
@@ -66,6 +66,32 @@ kubectl [command] [TYPE] [NAME] [flags]
...
+
+**`POD_NAMESPACE`環境変数**
+
+`POD_NAMESPACE`環境変数が設定されている場合、名前空間に属するリソースのCLI操作は、デフォルトで環境変数の値になります。これは、Podが名前空間に属するリソースであり、コマンドで名前空間が指定されていないためです。`kubectl api-resources`の出力を見て、リソースが名前空間に属するかどうかを判断してください。
+
+明示的に`--namespace <value>`を使用すると、この動作は上書きされます。
+
+**kubectlによるServiceAccountトークンの処理方法**
+
+以下の条件がすべて成立した場合、kubectlはクラスターで実行されているとみなします。
```
Comment (nasa9084) [inline content/ja/docs/reference/kubectl/_index.md:83]:
```suggestion
以下の条件がすべて成立した場合、kubectlはクラスター内で実行されているとみなします。
```

Comment (nasa9084) [reply content/ja/docs/reference/kubectl/_index.md:83]: btw, Although it's an issue in the English document, the previous section said the in-cluster execution condition were 3, but this list has 4 items...?

File: content/ja/docs/reference/kubectl/_index.md (hunk 1 above)
Comment (s-kawamura-w664) [reply content/ja/docs/reference/kubectl/_index.md]: That sentence seems to had been forgotten.
It was translated and added in ja.

**Comment by s-kawamura-w664** [comment]
@atoato88 @nasa9084 
I updated about your comments.
Could you please review again?

**Review by nasa9084 — COMMENTED** [review]
/approve

**Pull request merged**